### PR TITLE
postgres: do not copy and linearize received data when it is not going to be used

### PR DESCRIPTION
**Pull request by cpakulski**
Commit Message:
When the decoder analyzes postgres messages in order to produce statistics, the message body needs to be linearized and copied to the internal buffer.

From around 40 types of postgres messages, only about 8-10 are used to produce stats, but linearization and copying to internal buffer was done for all messages. Code in this PR checks if linearization and copying is needed.

Also, after linearization, the decoder stored the linearized message until the next message arrived instead of releasing the memory immediately after processing. Code in this PR releases memory after the message has been processed.

Additional Description:

Risk Level: Low

Testing: Adjusted unit test to verify that memory is released after processing the message.

Docs Changes: No

Release Notes: No

[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/filters/network/postgres_proxy/postgres_decoder.cc
Comment (dio) [inline source/extensions/filters/network/postgres_proxy/postgres_decoder.cc:264]: @cpakulski sorry, do we have tests to check this condition? To make sure, if `actions.empty()` we don't call `linearize` at all.

Comment (cpakulski) [reply source/extensions/filters/network/postgres_proxy/postgres_decoder.cc:264]: Added unit test to verify that.

**Comment by dio** [comment]
Looks good. Asking help from @zuercher for senior maintainer review. Thanks!

**Review by zuercher — APPROVED** [review]
lgtm

**Comment by dio** [comment]
@cpakulski, sorry, seems like docs build is hanging. See if merging main could help.

**Comment by cpakulski** [comment]
@zuercher I need your "lgtm" again. Nothing really changed from the last review. I had to remove one mock method because it was removed from the base class. Thanks!

**Review by zuercher — APPROVED** [review]

**Pull request merged**
